python/cudf/cudf/core/frame.py

```diff
@@ -658,7 +658,7 @@ def _copy_categories(self, other, include_index=True):
             ):
                 self._data[name] = build_categorical_column(
                     categories=other_col.categories,
-                    codes=col,
+                    codes=as_column(col.base_data, dtype=col.dtype),
```

Comment:
This feels super janky.

Reply:
What about it other than the verbosity? The alternative is to make this more expensive and use `col.data` and `col.mask` instead :(

Reply:
I think moreso that there's ambiguity of how offsets / size are handled for codes vs the CategoricalColumn itself. I.E. imagine wanting to construct a CategoricalColumn with an offset from a codes NumericalColumn that also has a different offset. That isn't possible with this.

Reply:
Removed this entirely from the PR